### PR TITLE
Add support to linestring functions

### PR DESCRIPTION
**Pull request by massilva**

### CODE REVIEW

File: geoalchemy2/functions.py [hunk 1]
```diff
@@ -312,6 +312,18 @@ def __init__(self, *args, **kwargs):
      'Returns a geometry that represents the shared portion of geomA and '
      'geomB. The geography implementation does a transform to geometry to do '
      'the intersection and then transform back to WGS84.'),
+   
+    ('ST_Line_Locate_Point', types.Geometry,
+     'Returns a locate point from geometryA into line.'),
```
Comment (elemoine) [inline geoalchemy2/functions.py]: The name should be `ST_LineLocatePoint` as in PostGIS. And please use the same description as the one in the PostGIS doc. See http://postgis.net/docs/ST_LineLocatePoint.html.

File: geoalchemy2/functions.py [hunk 2]
```diff
@@ -312,6 +312,18 @@ def __init__(self, *args, **kwargs):
...
+
+
+    ('ST_LineMerge', types.Geometry,
+     'Returns a geometry that represents that merge of lineStringA and '
+     'lineStringB.'),
```
Comment (elemoine) [inline geoalchemy2/functions.py]: See http://postgis.net/docs/ST_LineMerge.html for the description.

File: geoalchemy2/functions.py [hunk 3]
```diff
@@ -312,6 +312,18 @@ def __init__(self, *args, **kwargs):
...
+
+    ('ST_Line_Substring', types.Geometry,
```
Comment (elemoine) [inline geoalchemy2/functions.py]: `ST_LineSubstring`. See http://postgis.net/docs/ST_LineSubstring.html.

**Comment by elemoine** [comment]
Thanks for the patch. I have comments regarding the function names and descriptions.

Also, note that the Travis build fails because of code-style issues:

```
$ flake8 geoalchemy2 tests --ignore=E711
geoalchemy2/functions.py:315:1: W293 blank line contains whitespace
geoalchemy2/functions.py:325:80: E501 line too long (86 > 79 characters)
```

**Comment by massilva** [comment]
@elemoine I did the changes. See commit 7a73f1731cdd2b32d7265710c05c387feaaa7a7c

File: geoalchemy2/functions.py [hunk 4]
```diff
@@ -313,6 +313,30 @@ def __init__(self, *args, **kwargs):
      'geomB. The geography implementation does a transform to geometry to do '
      'the intersection and then transform back to WGS84.'),
 
+    ('ST_LineLocatePoint', types.Geometry,
+     'Returns a float between 0 and 1 representing the location of the closest'
```
Comment (elemoine) [inline geoalchemy2/functions.py]: Space missing after `closest`?

File: geoalchemy2/functions.py [hunk 5]
```diff
@@ -313,6 +313,30 @@ def __init__(self, *args, **kwargs):
...
+     'You can use the returned location to extract a Point '
+     '(ST_LineInterpolatePoint) or a substring (ST_LineSubstring).'
+     ''
+     'This is useful for approximating numbers of addresses'),
+
+    ('ST_LineMerge', types.Geometry,
+     'Returns a (set of) LineString(s) formed by sewing together the '
+     'constituent line work of a MULTILINESTRING.'),
+
+    ('ST_Line_Substring', types.Geometry,
```
Comment (elemoine) [inline geoalchemy2/functions.py]: `ST_LineSubstring`

**Comment by massilva** [comment]
@elemoine I did the changes. See commit 5ceeb2d297b954d668d7766835cd089b2832b9df

**Comment by elemoine** [comment]
The build still does not pass on Travis. See https://travis-ci.org/geoalchemy/geoalchemy2/jobs/218546767#L4914. You have a line that is too long.

**Comment by massilva** [comment]
The build is OK!

**Pull request merged**